### ISSUE TITLE
Beta: Uncaught Error: watch [...]/electron.asar/renderer ENOTDIR

### ISSUE THREAD

**Issue by pichfl** [felixrieseberg/ember-electron#182]
This error shows up when running `ember electron` as well as when running an app packaged with `ember electron:package`.

```
Uncaught Error: watch /Users/pichfl/Repositories/tron/node_modules/electron-prebuilt-compile/node_modules/electron/dist/Electron.app/Contents/Resources/electron.asar/renderer ENOTDIR
```

**Comment by jacobq** [comment]
If I put a breakpoint in reload.js on `fs.watch(dirname, { recursive: true }, () => window.location.reload());` I can then go over to the console and see that:

* `fs.statSync(dirname).isDirectory()` => `true`
* `fs.watch(dirname)` throws `ENOTDIR`

This is definitely a confusing error message.

**Comment by jacobq** [comment]
Seems like `__dirname` isn't what we expect it to be [here](https://github.com/felixrieseberg/ember-electron/blob/a1b1ccfed59be164513d965e9aca116bd815f8d6/app/electron/reload.js#L19). There's a comment that suggests it normally aligns with `./dist/` but debugger shows `./node_modules/electron/dist/resources/electron.asar/renderer`

**Comment by jacobq** [comment]
I'm not sure if this is right, but I was able to alleviate this with the following change:
https://github.com/jacobq/ember-electron/tree/fix-reload-watch-path

**Comment by pichfl** [comment]
Looks reasonable. Looking at the file, I just realised we have to find a more stable way to check if we are running in a testing environment given that not all ember projects use qunit, but thats something for a new ticket. (Will file)

I think you can start a pull request straight away.

**Comment by anulman** [comment]
Fixed in #188

**Comment by jacobq** [comment]
@anulman did you test ([this line](https://github.com/felixrieseberg/ember-electron/blob/refactor/blueprint/app/electron/reload.js#L81)) on linux? I now get:
```
Uncaught Error: watch /home/jquant/Documents/KFMT/Software/ember-electron-update-demo/assets ENOENT
    at exports._errnoException (util.js:1022)
    at FSWatcher.start (fs.js:1316)
    at Object.fs.watch (fs.js:1341)
    at watch (reload.js:22)
    at fs.stat (reload.js:79)
    at ELECTRON_ASAR.js:272
    at _combinedTickCallback (internal/process/next_tick.js:67)
    at process._tickCallback (internal/process/next_tick.js:98)
```

**Comment by anulman** [comment]
@jacobq Argh, no, didn't have access to that env at the time. Will have a solution shortly

**Comment by anulman** [comment]
@jacobq try now? https://github.com/felixrieseberg/ember-electron/blob/refactor/blueprint/app/electron/reload.js#L71-L87

**Comment by jacobq** [comment]
Just glanced at it and it looks reasonable, though I'm heading out for lunch right now; will check when I get back.

**Comment by jacobq** [comment]
Sorry, bad news: still no dice:
```
ember.debug.js:6001 DEBUG: -------------------------------
ember.debug.js:6001 DEBUG: Ember      : 2.12.0
ember.debug.js:6001 DEBUG: Ember Data : 2.12.0
ember.debug.js:6001 DEBUG: jQuery     : 3.2.0
ember.debug.js:6001 DEBUG: -------------------------------
/home/jquant/Documents/KFMT/Software/temp/junk/node_modules/devtron/api.js:5 Installing Devtron from /home/jquant/Documents/KFMT/Software/temp/junk/node_modules/devtron
fs.js:1318 Uncaught Error: watch /home/jquant/Documents/KFMT/Software/temp/junk/dist/ember/ember/assets ENOENT
    at exports._errnoException (util.js:1022)
    at FSWatcher.start (fs.js:1316)
    at Object.fs.watch (fs.js:1341)
    at watch (reload.js:21)
    at fs.stat (reload.js:78)
    at FSReqWrap.oncomplete (fs.js:112)
```

**Comment by anulman** [comment]
Derp, forgot to remove some nested paths. Try now?

**Comment by jacobq** [comment]
Works in `6491617a0cbce1611c9212cab0f1edc1cdfbdd4e` :+1:

**Comment by jacobq** [comment]
@anulman This seems to be broken again (at least on my Linux box; haven't tested other platforms yet)
It looks like `__dirname` in `reload.js` is `./tmp/electron_livereload-XXXXXXXX.tmp/ember` (note the **`/ember`**) but [here](https://github.com/felixrieseberg/ember-electron/blob/master/app/electron/reload.js#L65) we still add `ember` to the watch path. Does this work for you on macOS? I'd be surprised if this is platform-specific as it's not even related to the linux-only section of the code. 

If I replace that line with `let dirname = path.resolve(__dirname, '..') || process.cwd();` (i.e. going up a level) it works for me again.

**Comment by jacobq** [comment]
(FWIW the problem I mentioned again today with the extra `ember` folder in the path appears to be fixed by #207)
